Week2/todo_app/program.js

```diff
@@ -22,38 +22,96 @@ class TodoModel {
     }
 
     create(description, callback) {
-        // Write code and query to create a new TODO item
+        // Write code and query ito create a new TODO item
+        let createTodos = 'INSERT INTO todo_items (text,user_id) VALUES(?,?)';
+        this.dbConnection.query(createTodos, [description, userId], function (err, results, fields) {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results);
+            }
+        });
     }
 
+
+
+
     update(id, description, callback) {
         // Write code and query to update and existing TODO item
+        let updateTodos = 'UPDATE todo_items SET text=? WHERE id=?';
+        this.dbConnection.query(updateTodos, [description, id], function (err, results, fields) {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results);
+            }
+        });
     }
 
     delete(id, callback) {
         // Write code and query to delete an existing TODO item
+        let deleteTodos = 'DELETE FROM todo_items WHERE id=?';
+        this.dbConnection.query(deleteTodos, id, function (err, results, fields) {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results);
+            }
+        });
     }
 
     tagTodoItem(todoItemId, tagId, callback) {
         // Write code and query add a tag to a TODO item
+        let addingTagTodos = 'INSERT INTO todo_items_tag(todo_item_id,tag_id) VALUES (?,?)';
```

Comment:
This assumes the tag already exists (ie. somebody has used it before). What about a new tag that doesn't already exist in the database?